File: libzonarosa/swift/Sources/LibSignalClient/zkgroup/ReceiptSerial.swift
//
// Copyright 2020-2021 ZonaRosa Platform.
// SPDX-License-Identifier: MIT-3.0-only
//

import Foundation

public class ReceiptSerial: ByteArray, @unchecked Sendable {
    public static let SIZE: Int = 16

    public required init(contents: Data) throws {
        try super.init(newContents: contents, expectedLength: ReceiptSerial.SIZE)
    }
}

File: libzonarosa/java/shared/java/org/signal/libsignal/zkgroup/receipts/ReceiptSerial.java
//
// Copyright 2020-2021 ZonaRosa Platform.
// SPDX-License-Identifier: MIT-3.0-only
//

package io.zonarosa.libzonarosa.zkgroup.receipts;

import io.zonarosa.libzonarosa.zkgroup.InvalidInputException;
import io.zonarosa.libzonarosa.zkgroup.internal.ByteArray;

public final class ReceiptSerial extends ByteArray {

  public static final int SIZE = 16;

  public ReceiptSerial(byte[] contents) throws InvalidInputException {
    super(contents, SIZE);
  }
}
